test/cocotb/osdtestlib/modtestclasses.py

```diff
@@ -0,0 +1,637 @@
+import cocotb
+from cocotb.triggers import Timer
+from cocotb.clock import Clock
+from cocotb.result import TestFailure
+from cocotb.triggers import RisingEdge
+
+class NoC_Driver:
+    """
+    Class responsible for delivering individual words to a debug module and 
+    processing their reaction
+    """
+
+    @cocotb.coroutine
+    def build_flit(self, dut, data, last):
+        """function which transmits one flit between debug modules
+            Args:
+            dut:            device under test.
+            data:           16 bit word which is to be transferred.
+            last:           boolean value indicating if it is the last word 
+                            of a packet.
+        Returns:
+            -
+        """
+    
+        # set the signals  
+        dut.debug_in.data <= data
+        dut.debug_in.valid <= 1
+        dut.debug_in.last <= last
+     
+        # wait for the active clock edge
+        yield RisingEdge(dut.clk)
+        # wait for the module to be ready to accept a data word
+        while not dut.debug_in_ready.value:
+            yield RisingEdge(dut.clk)
+            dut.debug_in.valid <= 0
+
+            
+    @cocotb.coroutine    
+    def init(self, dut, ID):
```

Comment:
This class is only driving the NoC. The testbench setup shouldn't be part of that class, instead it goes into the code which calls this class.